pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -419,6 +427,17 @@ func validateName(name string) error {
 	if err := validation.IsQualifiedName(name); err != nil {
 		return fmt.Errorf("name is not a valid Kubernetes label name: %v", err)
 	}
+	// we expect the resource property names to be in this format `[PREFIX]/[CAPACITY_TYPE]-[RESOURCE_NAME]`.
+	if strings.HasPrefix(name, resourcePropertyNamePrefix) {
+		resourcePropertyName, _ := strings.CutPrefix(name, resourcePropertyNamePrefix)
+		segments := strings.Split(resourcePropertyName, "-")
+		if len(segments) != 2 || len(segments[0]) == 0 || len(segments[1]) == 0 {
```

Comment:
Had to replicate some validation logic from here https://github.com/Azure/fleet/blob/6d59196736446457aa9ff7b310fb150d5bca1c67/pkg/scheduler/framework/plugins/clusteraffinity/types.go#L43-L51, but it feels like the validation belongs here. Not removing the original validation as a safety measure. But since the webhook is fail close for CRP it never should reach the origin scheduler validation logic

Reply:
Hi Arvind! As mentioned in Teams, considering that webhook is optional and we will be having upstream only users as well, it might not be a good idea to assume that webhook will be able catch all the validation errors for us. As for the alert issue, it might be better to update the code to mark the error as a user one I think.

Reply:
Need to create a new user error object to be able to differentiate between the errors returned, so that we can bubble the error as a framework status at the top layer. Will handle it in a separate PR

Reply:
Need to classify errors as user errors https://github.com/Azure/fleet/blob/28e5a083bab5d437888b281cf0cf1c76a32ee0f1/pkg/scheduler/framework/plugins/clusteraffinity/types.go#L98-L110